### PR TITLE
RKE cli engine deployment of rancher via helm + expose via NodePort

### PR DESCRIPTION
**Pull request by bgentry-ntap**

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -15,6 +13,8 @@ import (
 	"github.com/spf13/viper"
 
 	"github.com/netapp/cake/pkg/config/events"
+	"github.com/netapp/cake/pkg/engines/rke"
+	rkecli "github.com/netapp/cake/pkg/engines/rke-cli"
```
Comment (jacobweinstock) [inline cmd/deploy.go:17]: i think best practice around package naming is a single word. and I think we've violated that all round, not just here. we probably will need a PR just to update package names :(

https://rakyll.org/style-packages/ 
https://blog.golang.org/package-names

File: pkg/engines/rke-cli/rke-cli.go [hunk 2]
```diff
@@ -158,7 +164,118 @@ func (c *MgmtCluster) CreatePermanent() error {
...
+
+	log.Infof("waiting for rancher to be ready")
+	args = []string{
+		"rollout",
+		"status",
+		"deploy/rancher",
+		fmt.Sprintf("--namespace=%s", namespace),
+		fmt.Sprintf("--kubeconfig=%s", kubeConfigFile),
+	}
+	cmd := exec.Command("kubectl", args...)
```
Comment (jacobweinstock) [inline pkg/engines/rke-cli/rke-cli.go:239]: you're using `cmds.GenericExecute`above but use `exec.Command` here. probably should pick one.

File: cmd/deploy.go (hunk 1 above)
Comment (bgentry-ntap) [reply cmd/deploy.go:17]: This was a poor name on my part, but just adjusted the alias to be a single word for golint. Definitely in favor of a package name cleanup

File: pkg/engines/rke-cli/rke-cli.go (hunk 2 above)
Comment (bgentry-ntap) [reply pkg/engines/rke-cli/rke-cli.go:239]: I use exec directly when I need the command object back, for example to wait here. I can switch to only using exec.

**Pull request merged**